File: app/src/main/java/Model/Offer/Offer.java
package Model.Offer;

import com.google.firebase.Timestamp;
import com.google.firebase.firestore.Blob;

import java.util.List;
import java.util.Map;

public class Offer {

    Timestamp startTime, endTime;
    String offerId;
    Blob offerPic;
    float discountPercentage;

    public Offer() {
    }

    public Timestamp getStartTime() {
        return startTime;
    }

    public void setStartTime(Timestamp startTime) {
        this.startTime = startTime;
    }

    public Timestamp getEndTime() {
        return endTime;
    }

    public void setEndTime(Timestamp endTime) {
        this.endTime = endTime;
    }

    public float getDiscountPercentage() {
        return discountPercentage;
    }

    public void setDiscountPercentage(float discountPercentage) {
        this.discountPercentage = discountPercentage;
    }

    public String getOfferId() {
        return offerId;
    }

    public void setOfferId(String offerId) {
        this.offerId = offerId;
    }

    public Blob getOfferPic() {
        return offerPic;
    }

    public void setOfferPic(Blob offerPic) {
        this.offerPic = offerPic;
    }
}

File: app/src/main/java/com/whiteside/cafe/model/Offer.kt
package com.whiteside.cafe.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.Blob
import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class Offer(
    var categoryName: String,
    var productId: String,
    var offerId: String,
    var startTime: Timestamp,
    var endTime: Timestamp,
    var offerPic: Blob,
    var discountPercentage: Float,
) {

    constructor() : this("", "", "",  Timestamp.now(),  Timestamp.now(), Blob.fromBytes(byteArrayOf()), 0f)

//    lateinit var categoryName: String
//    lateinit var productId: String
//    lateinit var offerId: String
//    lateinit var startTime: Timestamp
//    lateinit var endTime: Timestamp
//    lateinit var offerPic: Blob
//    var discountPercentage: Float? = null


}